Modules/System Test Libraries/Environment Information/src/EnvironmentInfoTestLibrary.Codeunit.al

```diff
@@ -19,30 +19,63 @@ codeunit 135094 "Environment Info Test Library"
     /// This functions should only be used for testing purposes.
     /// </remarks>
     /// <param name="EnableSandboxForTest">The value to be set to the testability sandbox flag.</param>
-    [Scope('OnPrem')]
     procedure SetTestabilitySandbox(EnableSandboxForTest: Boolean)
     begin
+        EnvironmentInformationImpl.SetTestMode(true);
         EnvironmentInformationImpl.SetTestabilitySandbox(EnableSandboxForTest);
     end;
 
+    /// <summary>
+    /// Sets the test mode flag.
+    /// </summary>
+    /// <remarks>
+    /// This functions should only be used for testing purposes.
+    /// </remarks>
+    /// <param name="EnableTests">The value to be set to the test flag.</param>
+    procedure SetTestability(EnableTests: Boolean)
+    begin
+        EnvironmentInformationImpl.SetTestMode(EnableTests);
+    end;
+
+
     /// <summary>
     /// Sets the testability SaaS flag.
     /// </summary>
     /// <remarks>
     /// This functions should only be used for testing purposes.
     /// </remarks>
     /// <param name="EnableSoftwareAsAServiceForTest">The value to be set to the testability SaaS flag.</param>
-    [Scope('OnPrem')]
     procedure SetTestabilitySoftwareAsAService(EnableSoftwareAsAServiceForTest: Boolean)
     begin
+        EnvironmentInformationImpl.SetTestMode(true);
         EnvironmentInformationImpl.SetTestabilitySoftwareAsAService(EnableSoftwareAsAServiceForTest);
     end;
 
+    /// <summary>
+    /// Sets the App ID that of the current application (for example, 'FIN' - Financials).
+    /// </summary>
+    /// <param name="NewAppId">The desired new App ID.</param>
+    procedure SetTestabilityAppId(NewAppId: Text)
+    begin
+        EnvironmentInformationImpl.SetTestMode(true);
+        EnvironmentInformationImpl.SetTestabilityAppId(NewAppId);
+    end;
+
+    /// <summary>
+    /// Sets the Environment Name that of the current Environment.
+    /// </summary>
+    /// <param name="NewEnvironmentName">The desired new Environment Name.</param>
+    procedure SetTestabilityEn(NewEnvironmentName: Text)
```

Comment:
I think the name of this procedure should be `SetTestabilityEnvironmentName`?

Reply:
Yes correct, i've fixed that...